net/tcp/Kconfig

```diff
@@ -51,11 +51,45 @@ config NET_TCPURGDATA
 		compiled in. Urgent data (out-of-band data) is a rarely used TCP feature
 		that is very seldom would be required.
 
-config NET_TCP_CONNS
-	int "Number of TCP/IP connections"
+config NET_TCP_PREALLOC_CONNS
+	int "Preallocated TCP/IP connections"
 	default 8
 	---help---
-		Maximum number of TCP/IP connections (all tasks)
+		Number of TCP/IP connections (all tasks).
+		
+		This number of connections will be pre-allocated during system boot.
+		If dynamic connections allocation is enabled, more connections may
+		be allocated at a later time, as the system needs them. Else this
+		will be the maximum number of connections available to the system
+		at all times.
+
+		Set to 0 to disable (and rely only on dynamic allocations).
+
+config NET_TCP_ALLOC_CONNS
+	int "Dynamic TCP/IP connections allocation"
+	default 0
+	---help---
+		Dynamic memory allocations for TCP/IP.
+
+		When set to 0 all dynamic allocations are disabled.
+
+		When set to 1 a new connection will be allocated every time,
+		and it will be free'd when no longer needed.
+
+		Setting this to 2 or more will allocate the connections in
+		batches (with batch size equal to this config). When a
+		connection is no longer needed, it will be returned to the
+		free connections pool, and it will never be deallocated!
+
+config NET_TCP_MAX_CONNS
```

Comment:
@anchao Each protocol may have a vastly different use, so different limits may be needed.

Also each connection struct has a different size, so for a given heap size you need different limits to take advantage of it all.

Reply:
emm ... since in this way, too many configurations will be added to each protocol, could we only changes the number of pre-allocated connections to achieve this feature?